### ISSUE TITLE
Update on Cleaning the ID column

### ISSUE THREAD

**Issue by jzywaaaaa** [jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus#1]
@epageperron 

I have cleaned the data to this extent with the [uploaded jupyter notebook](https://github.com/jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus/blob/main/Conll_clean_up.ipynb):

<img width="668" alt="image" src="https://user-images.githubusercontent.com/90368485/205128517-433ca9ad-8685-415c-82c8-ac94356d0640.png">

My next step is to fix the ID column. I tried running the [ATF converter](https://github.com/cdli-gh/atf2conll-convertor) on the [original data dump](https://github.com/cdli-gh/data/blob/master/cdliatf_unblocked.atf). 132,623 were successfully converted to conll files with the ID column desired and 1,911 errored out. The error message is "IDs generated are not unique\" and the entire list of files erroring out can be found here: [error_message.txt](https://github.com/jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus/files/10135175/error_message.txt).

Because the files erroring out only consist of a small portion of all files, I plan to continue doing this. Hopefully most data we need are in the successfully converted conll files!

**Comment by epageperron** [comment]
**Looks great !**

Yes I agree with you, there should only be between a few and none of the files you are working on that can't be converted from the original ATF dataset. 

1) I've done some spot checks in your error file and definitely, the problem is in the text, not in the converter. So once we fix the text, the converter will work appropriately. 

2) to be CDLI-CoNLL conform, you would need these headers: # ID	FORM	SEGM	XPOSTAG	HEAD	DEPREL	MISC, so you are missing DEPREL, and we also need to replace the content of HEAD ( and DEPREL) with underscores. We are basically discarding the syntactic information completely. 

3) Additionally, for the files to conform, we need to add the top comments, eg: 
```
#new_text=P100065
# ID	FORM	SEGM	XPOSTAG	HEAD	DEPREL	MISC 

```
You can grab the text number from the file name while the column names will always be the same.   
  
Once you are satisfied with your script, please fork it to the org Github :)

**Comment by jzywaaaaa** [comment]
Thanks for the feedback!

Yes I will add the text number in the end.

I thought that you said edge and deprel data can be removed because they are part of the syntax data? Or do you mean keep the columns themselves but remove the content of the columns and use underscores as placeholder?

**Comment by epageperron** [comment]
yes exactly remove the data but keep the columns :) replacing with underscores

**Comment by epageperron** [comment]
But no "edge" column.

**Comment by jzywaaaaa** [comment]
@epageperron 

After fixing the above-mentioned mistakes and dealing with some corner cases, I have been able to do the conversion:

e.g., P216736
original file:
<img width="778" alt="image" src="https://user-images.githubusercontent.com/90368485/205457904-700a7a2c-d440-4a0f-975b-aeddfb0d3018.png">


cleaned up file:
<img width="697" alt="image" src="https://user-images.githubusercontent.com/90368485/205457870-c8fa1574-8eb3-4078-99ab-b4264308f660.png">

 but when I tried running my script on all subcorpus files I realized that some of the royal subcorpus conll data have different number of rows than the conll file converted by the ATF converter:

e.g., P430122

This is the conll file produced by Chistian's tool:
<img width="837" alt="image" src="https://user-images.githubusercontent.com/90368485/205457986-d6413717-686f-4572-a7f9-4344055d9f64.png">

This is the conll file produced by the ATF converter:
<img width="413" alt="image" src="https://user-images.githubusercontent.com/90368485/205458046-b12cbab7-f9ff-4989-9ade-3b1592f55464.png">

As you can see, a.1.4.2 in the ATF converted file is split between row 7 and row 8 in the subcorpus file. I can fix this by manually comparing the two files, but I have not come up with a way to embed this process in my script and do it on all files. Do you have any suggestions for solving this problem?

**Comment by epageperron** [comment]
Ah yes ! If you can detect which ones don't align we can try conllmerge on them!

**Comment by epageperron** [comment]
https://github.com/acoli-repo/conll-merge conll

**Comment by jzywaaaaa** [comment]
ok l'll try detecting them first

**Comment by jzywaaaaa** [comment]
@epageperron 

I have successfully run my script on all 621 files Christian's tool generated. Unfortunately, 467 files do not have the same row number as that generated by the ATF converter and 2 files are among the ones the ATF converter failed to convert. 

You can find my code [here](https://github.com/jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus/blob/main/Conll_clean_up.py), the zip file of the successfully converted files [here](https://github.com/jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus/blob/main/clean_cdli_conll_files.zip), and a list of the error messages I asked my program to print here: [no_match_or_exist_error.txt](https://github.com/jzywaaaaa/Cleaning-Conll-Output-of-the-Royal-Subcorpus/files/10147842/no_match_or_exist_error.txt).

**Comment by epageperron** [comment]
Oh that's disappointing! However the files that are successfully converted look great! Just need to remove the data in the HEAD column and replace with underscores and those will be ready. 

  
Next step would be to try and run conll merge, after the columns are fixed but before updating the IDs I think.

**Comment by jzywaaaaa** [comment]
Just updated HEAD.

**Comment by epageperron** [comment]
Wonderful work, many thanks! :-) I will go over manually to fix a few little unexpected things and import the data in the database, that will be a small batch done ! :)

**Comment by jzywaaaaa** [comment]
sounds great! Thank you! Meanwhile I will work on conll merge

**Comment by epageperron** [comment]
Let's have a quick chat in #annotations when you have time, I think it wasn't easy to find the right parameters for conll-merge on the cdli stack for our needs so there is fiddling expertise in the group :)
